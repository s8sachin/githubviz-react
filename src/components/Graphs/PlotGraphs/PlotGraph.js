import React, { Component } from 'react';
import {curveCatmullRom} from 'd3-shape';
import { Grid, Row} from 'react-bootstrap';

import {
  XYPlot,
  XAxis,
  YAxis,
  HorizontalGridLines,
  VerticalGridLines,
  LineSeries
} from 'react-vis';


class PlotGraph extends Component {
  render() {
    return (
      <div>
          <Grid>
              <Row>
                <XYPlot
                    width={300}
                    height={300}>
                    <HorizontalGridLines />
                    <VerticalGridLines />
                    <XAxis title="X Axis" position="start"/>
                    <YAxis title="Y Axis"/>
                <LineSeries
                    className="first-series"
                    data={[
                    {x: 1, y: 3},
                    {x: 2, y: 5},
                    {x: 3, y: 15},
                    {x: 4, y: 12}
                    ]}/>
                <LineSeries
                    className="second-series"
                    data={null}/>
                <LineSeries
                    className="third-series"
                    curve={'curveMonotoneX'}
                    style={{
                    strokeDasharray: '2 2'
                    }}
                    data={[
                    {x: 1, y: 10},
                    {x: 2, y: 4},
                    {x: 3, y: 2},
                    {x: 4, y: 15}
                    ]}
                    strokeDasharray="7, 3"
                    />
                <LineSeries
                    className="fourth-series"
                    curve={curveCatmullRom.alpha(0.5)}
                    data={[
                    {x: 1, y: 7},
                    {x: 2, y: 11},
                    {x: 3, y: 9},
                    {x: 4, y: 2}
                    ]}/>
                </XYPlot>
                </Row>
            </Grid>
    </div>
    );
  }
}

export default PlotGraph;
