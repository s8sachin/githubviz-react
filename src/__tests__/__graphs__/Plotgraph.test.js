import React from 'react';
import { shallow,  configure } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import PlotGraph from '../../components/Graphs/PlotGraphs/PlotGraph';

configure({ adapter: new Adapter() });

describe('PlotGraph', () => {
it('should mount Header component', () => {
    const wrapper = shallow(<PlotGraph />)
    expect(toJson(wrapper)).toMatchSnapshot();
  })
})