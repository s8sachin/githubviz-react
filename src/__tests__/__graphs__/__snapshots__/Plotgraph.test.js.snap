// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlotGraph should mount Header component 1`] = `
<div>
  <Grid
    bsClass="container"
    componentClass="div"
    fluid={false}
  >
    <Row
      bsClass="row"
      componentClass="div"
    >
      <XYPlot
        className=""
        height={300}
        width={300}
      >
        <HorizontalGridLines
          attr="y"
          direction="horizontal"
        />
        <VerticalGridLines
          attr="x"
          direction="vertical"
        />
        <XAxis
          attr="x"
          attrAxis="y"
          orientation="bottom"
          position="start"
          title="X Axis"
        />
        <YAxis
          attr="y"
          attrAxis="x"
          orientation="left"
          title="Y Axis"
        />
        <LineSeries
          className="first-series"
          curve={null}
          data={
            Array [
              Object {
                "x": 1,
                "y": 3,
              },
              Object {
                "x": 2,
                "y": 5,
              },
              Object {
                "x": 3,
                "y": 15,
              },
              Object {
                "x": 4,
                "y": 12,
              },
            ]
          }
          getNull={[Function]}
          opacity={1}
          stack={false}
          strokeStyle="solid"
          style={Object {}}
        />
        <LineSeries
          className="second-series"
          curve={null}
          data={null}
          getNull={[Function]}
          opacity={1}
          stack={false}
          strokeStyle="solid"
          style={Object {}}
        />
        <LineSeries
          className="third-series"
          curve="curveMonotoneX"
          data={
            Array [
              Object {
                "x": 1,
                "y": 10,
              },
              Object {
                "x": 2,
                "y": 4,
              },
              Object {
                "x": 3,
                "y": 2,
              },
              Object {
                "x": 4,
                "y": 15,
              },
            ]
          }
          getNull={[Function]}
          opacity={1}
          stack={false}
          strokeDasharray="7, 3"
          strokeStyle="solid"
          style={
            Object {
              "strokeDasharray": "2 2",
            }
          }
        />
        <LineSeries
          className="fourth-series"
          curve={[Function]}
          data={
            Array [
              Object {
                "x": 1,
                "y": 7,
              },
              Object {
                "x": 2,
                "y": 11,
              },
              Object {
                "x": 3,
                "y": 9,
              },
              Object {
                "x": 4,
                "y": 2,
              },
            ]
          }
          getNull={[Function]}
          opacity={1}
          stack={false}
          strokeStyle="solid"
          style={Object {}}
        />
      </XYPlot>
    </Row>
  </Grid>
</div>
`;
